app/models/activity_type.rb

```diff
@@ -2,4 +2,36 @@ class ActivityType < ActiveRecord::Base
 
   validates :name,         presence: true, uniqueness: true
   validates :abbreviation, presence: true, uniqueness: true
+
+  after_destroy :invalidate_cache
+  after_save :invalidate_cache
+
+  def self.find(id)
+    Rails.cache.fetch("activity_types/#{id}", expires_in: 12.hours) do
+      super
+    end
+  end
+
+  def self.find_by(name)
+    Rails.cache.fetch("activity_types/#{name}", expires_in: 12.hours) do
+      super
+    end
+  end
+
+  def self.find_by(abbreviation)
```

Comment:
This wont work in rails - as this is the same as the code on line 15

Reply:
You may have to look at how to override this correctly. I assume it gets passed a hash with the key and value in it.